custom/lib_portal/models/res_partner.py

```diff
@@ -12,9 +12,12 @@ def _isMember(self):
     date_of_birth = fields.Date(string='Date of Birth', 
                                 default=fields.Date.context_today)
     date_of_death = fields.Date(string='Date of Death') 
-    languages_known = fields.Many2many('res.lang',string='Known Languages')
+    languages_known = fields.Many2many('res.lang', 'partner_lang_rel', 
+                                       string='Known Languages')
+	is_author = fields.Boolean(string='Is a Author', default=False, store=True)
+    is_publisher = fields.Boolean(string='Is a Publisher', default=False, 
+                                  store=True)
     is_member = fields.Char(compute='_isMember', string='Membership Status')
-   
-  
-   
+    book_ids = fields.One2many('product.template', 'author_id')
+    publisher_ids = fields.One2many('product.template', 'publisher_id')  
```

Comment:
the name of this field doesn't describe well its purpose

Reply:
_ids is used for the field name because of one2many fields.book_ids is to show the respective books written by authors in the kanban view when we open the author res partner view
and publisher_ids is to show the books published by the publishers in the kanban view of respective publisher res partner view